### PR TITLE
V2II2V Detection in sampling-LBR.

### PR DESCRIPTION
**Pull request by jestrang**
build.py updated with v2ii2v genkernel.

lbr.py = base code
common_lbr.py = stats
x86.py = list of v2ii2v triggering instructions

### CODE REVIEW

File: kernels/build.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def build_kernel(source, output=None, flags=''):
...
...xmm1' 'vcvtusi2ss %r8d,%xmm0,%xmm1' 'vcvtusi2ss %r8,%xmm0,%xmm1' 'vextractps $0x7e,%xmm0,%r8d' 'vmovd %r8d,%xmm0' 'vmovd %xmm0,%r8' 'vmovq %xmm0,%r8' 'vmovmskpd %xmm0,%r8d' 'vmovmskpd %ymm0,%r8d' 'vmovmskps %xmm0,%r8' 'vmovmskps %ymm0,%r8' 'vmovq %r8,%xmm0' 'vpcmpestri $0x7e,0x100(%rsp),%xmm0' 'vpcmpestri $0x7e,%xmm0,%xmm1' 'vpcmpestrm $0x7e,0x100(%rsp),%xmm0' 'vpcmpestrm $0x7e,%xmm0,%xmm1' 'vpcmpistri $0x7e,0x100(%rsp),%xmm0' 'vpcmpistri $0x7e,%xmm0,%xmm0' 'vpcmpistrm $0x7e,0x100(%rsp),%xmm0' 'vpcmpistrm $0x7e,%xmm0,%xmm0' 'vpextrb $0x7e,%xmm0,%r8d' 'vpextrd $0x7e,%xmm0,%r8d' 'vpextrq $0x7e,%xmm0,%r8' 'vpextrw $0x7e,%xmm0,%r8d' 'vpinsrb $0x7e,%r8d,%xmm0,%xmm1' 'vpinsrd $0x7e,%r8d,%xmm0,%xmm1' 'vpinsrq $0x7e,%r8,%xmm0,%xmm1' 'vpinsrw $0x7e,%r8d,%xmm0,%xmm1' 'vpmovmskb %xmm0,%r8d' 'vpmovmskb %ymm0,%r8d' 'vptest 0x100(%rsp),%xmm0' 'vptest 0x100(%rsp),%ymm0' 'vptest %xmm0,%xmm0' 'vptest %ymm0,%ymm0' 'vtestpd 0x100(%rsp),%xmm0' 'vtestpd 0x100(%rsp),%ymm0' 'vtestpd %xmm0,%xmm0' 'vtestpd %ymm0,%ymm0' 'vtestps 0x100(%rsp),%xmm0' 'vtestps 0x100(%rsp),%ymm0' 'vtestps %xmm0,%xmm0' 'vtestps %ymm0,%ymm0' 'vucomisd 0x100(%rsp),%xmm0' 'vucomisd %xmm0,%xmm0' 'vucomiss %xmm0,%xmm1' -n1 ",'v2ii2v')
```
Comment (amiri-khalil) [inline kernels/build.py:95]: Why did you put all the cases in? Is it a must? I believe a simple kernel with 1-2 cases is enough, just to ensure new stats are working when needed.

File: lbr/x86.py [hunk 2]
```diff
@@ -37,11 +37,33 @@
...
 LOAD_ANY  = r"[a-z0-9]+\s+[^,]*\("  # use is_mem_load()
 MOV       = r"v?mov"
-MOVS_ZX     = r"%s(s|zx)" % MOV
+MOVS_ZX   = r"%s(s|zx)" % MOV
 STORE     = r"\s+\S+\s+[^\(\),]+,"  # use is_mem_store()
-TEST_CMP  = r"(test|cmp).?\s"
-
-MEM_IDX = r"\((%[a-z0-9]+)?,%[a-z0-9]+,?(1|2|4|8)?\)"
+TEST_CMP  = r"(test|cmp).?\s" 
+cvt_suff  = ['sd2si', 'si2sd', 'si2ss', 'ss2si', 'tss2si']
```
Comment (amiri-khalil) [inline lbr/x86.py:43]: This part of the script is related to common instructions mnemonics, its better to write the V2IIV list elsewhere and not mix it in, probably write it at the end of the file.

File: kernels/build.py (hunk 1 above)
Comment (jestrang) [reply kernels/build.py]: reduced testcases

File: lbr/x86.py (hunk 2 above)
Comment (jestrang) [reply lbr/x86.py:43]: moved to bottom

File: kernels/build.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def build_kernel(source, output=None, flags=''):
     gen_kernel("-i 'movdqa %xmm1,%xmm2' 'andps %xmm3, %xmm2' NOP -n 14", 'v-mov-op-nop')
     gen_kernel("-i 'movups (%rsp),%xmm1' NOP 'andps %xmm2, %xmm1' -n 14", 'v-ld-nop-op')
     gen_kernel("-i 'movdqa %xmm1,%xmm2' NOP 'andps %xmm3, %xmm2' -n 14", 'v-mov-nop-op')
-
+    gen_kernel("-i 'aeskeygenassist $0x7e,(%rsp),%xmm0' 'aeskeygenassist $0x7e,%xmm0,%xmm1' 'cmpxchg16b 0x100(%rsp)' 'comisd 0x100(%rsp),%xmm0' 'comisd %xmm0,%xmm1' 'comiss 0x100(%rsp),%xmm0' 'comiss %xmm0,%xmm1' 'cvtsd2si 0x100(%rsp),%r8' 'cvtsd2si %xmm0,%r8d' 'cvtsi2sd %r8d,%xmm0' 'cvtsi2ss %r8d,%xmm0' 'cvtss2si 0x100(%rsp),%r8' 'cvtss2si %xmm0,%r8d' 'cvtss2si %xmm0,%r8' 'cvttsd2si 0x100(%rsp),%r8' 'cvttsd2si 0x100(%rsp),%r8d' 'cvttsd2si %xmm0,%r8' 'cvttsd2si %xmm0,%r8d' 'cvttss2si 0x100(%rsp),%r8' 'cvttss2si 0x100(%rsp),%r8d' -n1 ",'v2ii2v')
```
Comment (amiri-khalil) [inline kernels/build.py:95]: Can you do 2 kernels? One V2I and one I2V? with one single case in each kernel?

File: lbr/x86.py (hunk 2 above)
Comment (amiri-khalil) [reply lbr/x86.py:43]: Ok, but please restore the TEST_CMP and MEM_IDX to their original place.
Also, cvt_suff and suff are used only in the V2II2V list so declare them right above it and not up there.

**Review by amiri-khalil — APPROVED** [review]

**Pull request merged**